src/clj/puppetlabs/services/jruby/jruby_core.clj

```diff
@@ -183,6 +184,12 @@
   [jruby-instance :- (schema/pred jruby-schemas/jruby-instance?)]
   @(:state jruby-instance))
 
+(schema/defn register-event-handler
+  "Register the callback function by adding it to the event callbacks atom."
+  [event-callbacks :- Atom
+   callback-fn :- IFn]
+  (swap! event-callbacks conj callback-fn))
```

Comment:
I think this is the same basic conversation as what I was commenting on above, but yeah I think we should still have a `register-event-handler` fn that hides the implementation details (atom), but that it could probably just be a regular function that takes the pool as an argument, rather than needing to be a service function.

As far as use cases go, there is a service in pe-puppet-server-extensions that registers a callback and uses it to compute all of the JRuby metrics in PE.  We might have to make the OSS puppet server jruby service have a 'register-event-handler' service function that closes over its pool object, so that the pe service can call that.